Add FreshPorts report (#345)
freebsd/freebsd-doc: website/content/en/status/report-2024-01-2024-03/freshports.adoc

=== FreshPorts: Notification of new packages

Links: +

> link:https://freshports..org[FreshPorts] URL: link:freshports.org[] +

Reviewer: Extra dot before "org".

> link:https://freshports..org[FreshPorts] URL: link:freshports.org[] +

Reviewer 2: ```suggestion
link:https://freshports.org/[FreshPorts] URL: link:https://freshports.org/[] +
```

- URL: link:https: (the weird norm)
- plus the correction from @pauamma

link:https://news.freshports.org/[FreshPorts blog] URL: link:https://news.freshports.org/[]

Contact: Dan Langille <dvl@FreeBSD.org>

FreshPorts and FreshSource have reported upon FreeBSD commits for 20 years.
They cover all commits, not just ports.

FreshPorts tracks the commits and extracts data from the port Makefiles to create a database of information useful to both port maintainers and port users.

For example, link:https://www.freshports.org/security/acme.sh/#packages[] shows the history of the package:security/acme.sh[] port, back to its creation in May 2017.
Also available are dependencies, flavors, configuration options, and available packages.
All of this is useful for both users and developers of ports.

==== Notification: New Package Available

One of the original features of FreshPorts is notification of ports updates. You

> can create a list of ports and receive notifcations about those ports. This

Reviewer: s/notifcations/notifications/

> new feature can also notify when a new pacakge is available for that port.

Reviewer: s/pacakge/package/

> One of the original features of FreshPorts is notification of ports updates. You
> can create a list of ports and receive notifcations about those ports. This
> new feature can also notify when a new pacakge is available for that port.

Reviewer 2: ```suggestion
One of the original features of FreshPorts is notification of ports updates.
You can create a list of ports and receive notifications about those ports.
This new feature can also notify when a new package is available for that port.
```

- one sentence per line
- plus the corrections from @pauamma

> The use case: a known security vuln has been patched. FreshPorts will tell
> you the port has been patched, and then you wait for the package.

Reviewer 2: ```suggestion
The use case: a known security vuln has been patched.
FreshPorts will tell you the port has been patched, and then you wait for the package.
```

This new feature will tell you when that package is available.

Details at:

* link:https://github.com/FreshPorts/freshports/issues/542[]

==== Help Needed

It has been over 22 years since FreshPorts started.
Others must take over eventually.
I’ve started that process recently.
There are several aspects to FreshPorts:

* FreeBSD admin (updating the OS and packages)
* front end code (website - mostly PHP)
* back end code (commit processing - Perl, Python, shell)
* database design (PostgreSQL).

The database does not change very often and requires little maintenance compared to the applications and OS.
The website pretty much runs itself.
From time to time, a change to the FreeBSD ports infrastructure breaks something or requires a modification, but there is rarely any urgency to fix that.
This is not a huge time commitment.
There is a lot of learning.
While not a complex application, FreshPorts is also not trivial.

> To contribute, please join the
> link:https://lists.freshports.org/mailman/listinfo/freshports-coders[]
> mailing list and let us know what you'd like to help with.

Reviewer 2: ```suggestion
To contribute, please join the link:https://lists.freshports.org/mailman/listinfo/freshports-coders[] mailing list and let us know what you'd like to help with.
```
